Cleanup, Profiles, Bugfix for #10

asbachb opened the pull request:
- Removed some unused imports
- Introduced `release` profile (which is not activated by default) for easier local build 
- Added test case and fixed bug with double slash URLs (see #10)

@fzakaria sorry to bother you again. But could you please have a look.

fzakaria commented:
Can you provide some info you've validated the changes.
Show it compiles etc...

If you want to submit GitHub action... that would be amazing.

fzakaria commented on pom.xml:
```diff
@@ -70,6 +70,12 @@
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>5.6.0</version>
```
I would prefer these POM changes in a separate PR.

fzakaria commented on src/main/java/org/nixos/mvn2nix/URIBuilder.java:
```diff
@@ -0,0 +1,37 @@
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public final class URIBuilder {
```
Do we really need the copyright?

asbachb replied:
I don't need it I just added it, because they're present in the other classes. If we can remove it, I'd prefer to remove it as well.

fzakaria commented on src/test/java/org/nixos/mvn2nix/URIBuilderTest.java:
```diff
@@ -0,0 +1,55 @@
+
+	@Test
+	public void shouldBuildURI() throws Exception {
+		assertEquals(
```
what about more than 2 forward slashes.
What about just a base with additional slashes but not in the suffix?

asbachb replied:
I can add these test cases as well. So my idea was to add test cases for the cases reported in the bug.

fzakaria commented on src/main/java/org/nixos/mvn2nix/URIBuilder.java:
```diff
@@ -0,0 +1,37 @@
+	private URIBuilder() {
+	}
+	
+	public static URI build(String base, URI baseSuffix) throws URISyntaxException {
```
Comment this method please and lets make it not throw? Maybe Optional return?
I'm not a fan of checked exceptions in Java.

asbachb replied:
Just to clarify my intentions: I wanted to extract the code which caused #10 in order to write a test and validate if the fix works without refactoring the original code.

I hate that `URISyntaxException` in general as well. On the other hand what do we win with an `Optional` here? We need more handling in this class and the handling in caller is as long as before.

fzakaria reviewed (commented):
I have some feedback but I cant tell if the pom changes are all OK.
Please update with more info.

asbachb commented:
> Can you provide some info you've validated the changes. Show it compiles etc...

```
 ✘ asbachb@nixos-t14s  ~/dev/extsrc/mvn2nix-maven-plugin   develop  git log -n1 | cat
commit d96d69f3f10d40a2c22a18087775ef5828f7a173
Author: Benjamin Asbach <asbachb@users.noreply.github.com>
Date:   Mon Aug 1 21:32:31 2022 +0400

    feature: Added github action to build and test `mvn2nix-maven-plugin` via apache maven
 asbachb@nixos-t14s  ~/dev/extsrc/mvn2nix-maven-plugin   develop  mvn clean package
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------< org.nixos.mvn2nix:mvn2nix-maven-plugin >---------------
[INFO] Building mvn2nix 1.2.1
[INFO] ----------------------------[ maven-plugin ]----------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ mvn2nix-maven-plugin ---
[INFO] Deleting /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ mvn2nix-maven-plugin ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ mvn2nix-maven-plugin ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/target/classes
[INFO] /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/src/main/java/org/nixos/mvn2nix/Mvn2NixMojo.java: /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/src/main/java/org/nixos/mvn2nix/Mvn2NixMojo.java uses or overrides a deprecated API that is marked for removal.
[INFO] /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/src/main/java/org/nixos/mvn2nix/Mvn2NixMojo.java: Recompile with -Xlint:removal for details.
[INFO]
[INFO] --- maven-plugin-plugin:3.2:descriptor (default-descriptor) @ mvn2nix-maven-plugin ---
[INFO] Using 'UTF-8' encoding to read mojo metadata.
[INFO] Applying mojo extractor for language: java
[INFO] Mojo extractor for language: java found 0 mojo descriptors.
[INFO] Applying mojo extractor for language: bsh
[INFO] Mojo extractor for language: bsh found 0 mojo descriptors.
[INFO] Applying mojo extractor for language: java-annotations
[INFO] Mojo extractor for language: java-annotations found 1 mojo descriptors.
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ mvn2nix-maven-plugin ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ mvn2nix-maven-plugin ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ mvn2nix-maven-plugin ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.nixos.mvn2nix.URIBuilderTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.035 s - in org.nixos.mvn2nix.URIBuilderTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ mvn2nix-maven-plugin ---
[INFO] Building jar: /home/asbachb/dev/extsrc/mvn2nix-maven-plugin/target/mvn2nix-maven-plugin-1.2.1.jar
[INFO]
[INFO] --- maven-plugin-plugin:3.2:addPluginArtifactMetadata (default-addPluginArtifactMetadata) @ mvn2nix-maven-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.942 s
[INFO] Finished at: 2022-08-01T21:52:24+04:00
[INFO] ------------------------------------------------------------------------
```

> If you want to submit GitHub action... that would be amazing.

Added github action as suggested. Essentially used template from github with some minor adjustments: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven as I never configured one before ;)
